include/UniqueTable.hpp

```diff
@@ -0,0 +1,365 @@
+/*
+ * This file is part of the JKQ DD Package which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum_dd/ for more information.
+ */
+
+#ifndef DDpackage_UNIQUETABLE_HPP
+#define DDpackage_UNIQUETABLE_HPP
+
+#include "ComplexNumbers.hpp"
+#include "Definitions.hpp"
+#include "Edge.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <forward_list>
+#include <iostream>
+#include <limits>
+#include <numeric>
+#include <stack>
+#include <stdexcept>
+#include <vector>
+
+namespace dd {
+
+    /// Data structure for providing and uniquely storing DD nodes
+    /// \tparam Node class of nodes to provide/store
+    /// \tparam NBUCKET number of hash buckets to use (has to be a power of two)
+    /// \tparam INITIAL_ALLOCATION_SIZE number if nodes initially allocated
+    /// \tparam GROWTH_PERCENTAGE percentage that the allocations' size shall grow over time
+    /// \tparam INITIAL_GC_LIMIT number of nodes initially used as garbage collection threshold
+    /// \tparam GC_INCREMENT absolute number of nodes to increase the garbage collection threshold after garbage collection has been performed
+    template<class Node, std::size_t NBUCKET = 32768, std::size_t INITIAL_ALLOCATION_SIZE = 2048, std::size_t GROWTH_PERCENTAGE = 150, std::size_t INITIAL_GC_LIMIT = 250000, std::size_t GC_INCREMENT = 0>
```

Comment:
Do we need the option to grow by fractions? An integer `GROW_FACTOR` which defaults to 2 seems sufficient to me.

Reply:
See above. `GROWTH_FACTOR=2` should be sufficient.